src/initial_conditions/initial_conditions.jl

```diff
@@ -377,7 +380,24 @@ function overwrite_initial_conditions!(
     thermo_params,
 )
     extrapolation_bc = (Intp.Periodic(), Intp.Flat(), Intp.Flat())
-    file_path = weather_model_data_path(initial_condition.start_date)
+
+    # Extract face coordinates and compute center midpoints
+    z_coords = Fields.axes(Y.c).grid.vertical_grid.topology.mesh.faces
+
+    # Compute center coordinates as midpoints between faces
+    face_z_values = [z.z for z in z_coords]
+    center_z_values = [
+        (face_z_values[i] + face_z_values[i + 1]) / 2 for
+        i in 1:(length(face_z_values) - 1)
+    ]
+
+    target_levels = Array(center_z_values)
```

Comment:
```suggestion
    target_levels = Fields.field2array(Fields.coordinate_field(Y.c).z)[:, 1]
```
Is maybe the shortest fix but is a little chaotic with the indexing

Reply:
this is much cleaner but doesn't work on the GPU, looking for alternatives